crates/solver/src/settlement/settlement_encoder.rs

```diff
@@ -405,17 +405,16 @@ impl SettlementEncoder {
         Ok(execution)
     }
 
-    pub fn append_to_execution_plan(&mut self, interaction: impl Interaction + 'static) {
+    pub fn append_to_execution_plan(&mut self, interaction: Arc<dyn Interaction>) {
```

Comment:
The majority of changes in this diff originates from here which was needed to to make the interactions in the settlement clonable.

Reply:
Not directly related to this PR. It feels like we shouldn't store `dyn Interaction` in most cases. Instead we should store `EncodedInteraction`, which is a simple cloneable type. You could consider implementing something like this as a separate PR to make this one simpler.

Reply:
Makes sense. Will look into that.

Reply:
This is a problem with existing code so it's not blocking for this PR btw. Just wanted to mention it.

Reply:
The reason we store `dyn Interaction` is that we need each interaction to be able to decide how it interacts with others. For example WETH interaction "collapsing" logic is part of the `Interaction` trait and can't be done with an `EncodedInteraction`.